### PR TITLE
feat: add support for Angular v11 (follow up)

### PR DESCRIPTION
**Pull request by JohannesHoppe**
In retrospective, it wasn't my best idea to introduce the `--configuration` param. 😅 

* `configuration` is misleading as it might be confused with `configurations`.  @alan-agius4 already pointed this out here: https://github.com/angular/angularfire/issues/2063#issuecomment-570518883
* meanwhile, the CLI got configurations for `ng deploy` commands (which is NOT meant to be the configuration for the build as we did) @george43g pointed this out here: https://github.com/angular-schule/ngx-deploy-starter/issues/1#issuecomment-605646481

I think we want some kind of deprecation of our `--configuration` param since it's colliding with the designated param. We already have a breaking change due to the new stable deploy api. So let's clean everything up now.

See also
https://github.com/angular/angularfire/pull/2281

follow up of: https://github.com/angular-schule/angular-cli-ghpages/pull/118

FYI @beeman @dianjuar @mgechev

### CODE REVIEW

**Comment by beeman** [comment]
@JohannesHoppe I think it's a good call to clean it up! Happy to jump in to help, or test it locally!

**Comment by JohannesHoppe** [comment]
Great, so let's enable the same kind of stuff like in [angularfire](https://github.com/angular/angularfire/blob/master/docs/deploy/getting-started.md#working-with-multiple-firebase-projects). 🙂 

I wonder what we do with people who now use `ng deploy configuration=production`. The correct command would then be `ng deploy --buildTarget=production`. But I cannot introduce an alias for a transition period because `--configuration` has a new meaning. Do you think it is sufficient to write the following text in red letters?

**WARNING: The `configuration` option for specifying the build has been renamed to `buildTarget`. Please change the parameter (eg. rename `ng deploy --configuration=production` to  `ng deploy --buildTarget=production`). If you really want to adjust the deployment configuration, you can ignore this message.** 

Or do you have a better idea?

**Comment by beeman** [comment]
@JohannesHoppe I think that's a great solution to move away from the existing variable. Especially when you release it under a new major version folks can expect a breaking change :)

**Comment by dianjuar** [comment]
@beeman I couldn't agree more.

- Release those changes under a new major version
- The warning that you mention on the changelog 

I think that should be enough.

**Comment by dianjuar** [comment]
I think I should do the same on [ngx-deploy-npm](https://github.com/bikecoders/ngx-deploy-npm), following up all your comments and links sounds the way to go.

File: README.md [hunk 1]
```diff
@@ -196,7 +197,7 @@ This command has no effect if the option `--no-build` option is active.
 
 Skip build process during deployment.
 This can be used when you are sure that you haven't changed anything and want to deploy with the latest artifact.
-This command causes the `--configuration` setting to have no effect. -->
+This command causes the `--configuration` setting to have no effect.
```
Comment (fmalcher) [inline README.md]:
```suggestion
This command causes the `--buildTarget` setting to have no effect.
```

No?

File: src/deploy/schema.json [hunk 2]
```diff
@@ -7,11 +7,9 @@
       "type": "string",
       "description": "Base url for the application being built. Same as `ng build --base-href=/XXX/`."
     },
-    "configuration": {
+    "buildTarget": {
       "type": "string",
-      "default": "production",
-      "description": "A named build target, as specified in the `configurations` section of angular.json. Each named target is accompanied by a configuration of option defaults for that target. Same as `ng build --configuration=XXX`.",
-      "alias": "c"
+      "description": "A named build target, as specified in the `configurations` section of angular.json . Each named target is accompanied by a configuration of option defaults for that target. This is equivalent as calling the command `ng build --configuration=XXX`."
```
Comment (fmalcher) [inline src/deploy/schema.json:12]:
```suggestion
      "description": "A named build target, as specified in the `configurations` section of angular.json. Each named target is accompanied by a configuration of option defaults for that target. This is equivalent to calling the command `ng build --configuration=XXX`."
```

File: README.md [hunk 3]
```diff
@@ -36,7 +36,29 @@
...
+ng deploy --build-target=test
+```
+
+If you use the old syntax, you will probably receive the following error:
+
+> An unhandled exception occurred: Configuration 'test' is not set in the workspace.
+
+<br>
+
+**🐙 GitHub Actions** is now a first citizen alongside Travis CI and CircleCi. The token `GITHUB_TOKEN` is now supported. Learn everything you need to know in the following article.
```
Comment (fmalcher) [inline README.md:61]:
```suggestion
**🐙 GitHub Actions** is now properly supported alongside Travis CI and CircleCi. The token `GITHUB_TOKEN` is also supported. Learn everything you need to know in the following article.
```

The Google guide for inclusive documentation discourages the use of "first-class citizen": https://developers.google.com/style/inclusive-documentation#features-and-users

File: docs/README_contributors.md [hunk 4]
```diff
@@ -154,16 +157,20 @@ Use VSCode and debug the task `Launch Standalone Program`.
...
 npm publish dist
 npm dist-tag add angular-cli-ghpages@0.6.0-rc.0 next
 ```
 
-## Usage of Prettier Formatter
+## Keeping track of all the forks
+
+[ngx-deploy-starter](https://github.com/angular-schule/ngx-deploy-starter/) and
+[angular-cli-ghpages](https://github.com/angular-schule/angular-cli-ghpages/) (both developed by Johannes Hoppe) are follow-up projects of the depcrecated [ngx-gh demo](https://github.com/mgechev/ngx-gh).
+This project was a follow-up of the deploy shematics from the [angularfire](https://github.com/angular/angularfire/) project.
```
Comment (fmalcher) [inline docs/README_contributors.md:171]:
```suggestion
This project was a follow-up of the deploy schematics from the [angularfire](https://github.com/angular/angularfire/) project.
```

File: docs/README_contributors.md [hunk 5]
```diff
@@ -154,16 +157,20 @@ Use VSCode and debug the task `Launch Standalone Program`.
...
 
-Just execute `npx prettier --write '**/*'` and the code is formated automatically.
-Please ignore the errors for now. ([error] No parser could be inferred for file)
+To stay in sync with the stuff that the Angular team is doing, we want to keep an eye on the following files
```
Comment (fmalcher) [inline docs/README_contributors.md:173]:
```suggestion
To stay in sync with the stuff the Angular team is doing, you might want to keep an eye on the following files:
```

File: docs/README_contributors.md [hunk 6]
```diff
@@ -154,16 +157,20 @@ Use VSCode and debug the task `Launch Standalone Program`.
...
 npm run test
 npm publish dist
 npm dist-tag add angular-cli-ghpages@0.6.0-rc.0 next
 ```
 
-## Usage of Prettier Formatter
+## Keeping track of all the forks
+
+[ngx-deploy-starter](https://github.com/angular-schule/ngx-deploy-starter/) and
+[angular-cli-ghpages](https://github.com/angular-schule/angular-cli-ghpages/) (both developed by Johannes Hoppe) are follow-up projects of the depcrecated [ngx-gh demo](https://github.com/mgechev/ngx-gh).
```
Comment (fmalcher) [inline docs/README_contributors.md:170]:
```suggestion
[angular-cli-ghpages](https://github.com/angular-schule/angular-cli-ghpages/) (both developed by Johannes Hoppe) are follow-up projects of the deprecated [ngx-gh demo](https://github.com/mgechev/ngx-gh).
```

File: docs/README_contributors.md [hunk 7]
```diff
@@ -28,15 +27,19 @@ This may be useful when you want to try the latest non-published version of this
 
 Follow the instructions for [checking and updating the Angular CLI version](#angular-cli) and then link the package.
 
-### 1. Angular CLI
+### 1. Optional: Latest Angular version
 
-1. Install the next version of the Angular CLI.
+This builder requires the method `getTargetOptions()` from the Angular DevKit which was introduced [here](https://github.com/angular/angular-cli/pull/13825/files).
+All Angular projects with Angular 9 and greater are supposed to be compatible. (Actually it works with some versions of 8.x too, but you want to be up to date anyway, don't you?)
+Execute the next three steps, if your test-project is still older.
```
Comment (fmalcher) [inline docs/README_contributors.md:34]:
```suggestion
Execute the next three steps, if your test project is still older.
```

File: README.md [hunk 8]
```diff
@@ -172,19 +194,36 @@ ng deploy --cname=example.org
...
-  - `ng deploy` – Angular project is build in production mode
-  - `ng deploy --configuration=test` – Angular project is using the configuration `test` (this configuration must exist in the `angular.json` file)
+  - `ng deploy` – Angular project is build in `production` mode
+  - `ng deploy --build-target=test` – Angular project is using the build configuration `test` (this configuration must exist in the `angular.json` file)
+
+If no buildTarget is set, the `production` build of the default project will be chosen.
+The `buildTarget` simply points to an existing build configuration for your project, as specified in the `configurations` section of `angular.json`.
+Most projects have a default configuration and a production configuration (commonly activated by using the `--prod` flag) but it is possible to specify as many build configurations as needed.
+
+This is equivalent as calling the command `ng build --configuration=XXX`."
```
Comment (fmalcher) [inline README.md:209]:
```suggestion
This is equivalent to calling the command `ng build --configuration=XXX`.
```

File: README.md [hunk 9]
```diff
@@ -172,19 +194,36 @@ ng deploy --cname=example.org
...
+#### --build-target <a name="build-target"></a>
 
 - **optional**
-- Alias: `-c`
-- Default: `production` (string)
+- Default: `undefined` (string)
 - Example:
-  - `ng deploy` – Angular project is build in production mode
-  - `ng deploy --configuration=test` – Angular project is using the configuration `test` (this configuration must exist in the `angular.json` file)
+  - `ng deploy` – Angular project is build in `production` mode
```
Comment (fmalcher) [inline README.md:202]:
```suggestion
  - `ng deploy` – Angular project is built in `production` mode
```

File: README.md [hunk 10]
```diff
@@ -172,19 +194,36 @@ ng deploy --cname=example.org
...
+  - `ng deploy --build-target=test` – Angular project is using the build configuration `test` (this configuration must exist in the `angular.json` file)
+
+If no buildTarget is set, the `production` build of the default project will be chosen.
```
Comment (fmalcher) [inline README.md:205]:
```suggestion
If no `buildTarget` is set, the `production` build of the default project will be chosen.
```

File: README.md (hunk 1 above)
Comment (JohannesHoppe) [reply README.md]: Thanks!

**Comment by JohannesHoppe** [comment]
🎉 

Gentleman, I have pushed v1.0.0-rc.1 to NPM. Can you please give it a try and give me feedback if everything is fine?

I already left the tag to `latest`, so that it doesn't break for all people anymore.

File: README.md (hunk 3 above)
Comment (JohannesHoppe) [reply README.md:61]: Good to know! 👍

**Comment by beeman** [comment]
Thanks a lot for publishing the RC @JohannesHoppe! I just did a test and it seems to work like a charm!

**Comment by JohannesHoppe** [comment]
@beeman Thanks for the fast feedback. I will merge this PR to have the README up to date. I have some more ideas for rc2, so I would be happy for your continued support. 👍

**Comment by beeman** [comment]
> @beeman Thanks for the fast feedback. I will merge this PR to have the README up to date. I have some more ideas for rc2, so I would be happy for your continued support. 👍

For sure! I'm using it in several projects I'm currently working on so feel free to ping me when there is something to test 👍

**Comment by inpercima** [comment]
I just did a test too and it works. Thanks a lot.

But I do got some warnings on installing with yarn.

```
warning " > angular-cli-ghpages@1.0.0-rc.1" has unmet peer dependency "@angular-devkit/architect@>= 0.900 < 0.1200".
warning " > angular-cli-ghpages@1.0.0-rc.1" has unmet peer dependency "@angular-devkit/core@^9.0.0 || ^10.0.0 || ^11.0.0".
warning " > angular-cli-ghpages@1.0.0-rc.1" has unmet peer dependency "@angular-devkit/schematics@^9.0.0 || ^10.0.0 || ^11.0.0".
```

Not sure why. The dependencies are installed.

**Comment by JohannesHoppe** [comment]
🤔

**Comment by beeman** [comment]
I'm seeing the same. Here's a repro:

```
$ git clone git@github.com:beeman/angular-cli-ghpages-test.git
$ cd angular-cli-ghpages-test
$ yarn
yarn install v1.22.10
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning " > angular-cli-ghpages@1.0.0-rc.1" has unmet peer dependency "@angular-devkit/architect@>= 0.900 < 0.1200".
warning " > angular-cli-ghpages@1.0.0-rc.1" has unmet peer dependency "@angular-devkit/core@^9.0.0 || ^10.0.0 || ^11.0.0".
warning " > angular-cli-ghpages@1.0.0-rc.1" has unmet peer dependency "@angular-devkit/schematics@^9.0.0 || ^10.0.0 || ^11.0.0".
[4/4] 🔨  Building fresh packages...
✨  Done in 65.31s.
```

**Pull request merged**